include/circt/Dialect/Calyx/CalyxLoweringUtils.h

```diff
@@ -779,6 +779,28 @@ class BuildCallInstance : public calyx::FuncOpPartialLoweringPattern {
   ComponentOp getCallComponent(mlir::func::CallOp callOp) const;
```

Comment:
Nice, looking much better; I've added a few smaller comments. I am curious if you've verified at all the output of this code besides manually checking it, e.g., have you run the emitted code through the native Calyx compiler and checked the outputs via some backend like Icarus or SysVerilog?

Reply:
I went to test in Calyx and still trying to debug. I have a `oge` test case and the lowered result is:
```mlir
    group bb0_0 {
      std_compareFN_0.left = 32'd1;
      std_compareFN_0.right = 32'd9;
      std_compareFN_0.signaling = 1'b1;
      compare_port_0_reg.write_en = std_compareFN_0.done;
      compare_port_0_reg.in = (std_compareFN_0.lt == 1'd0) ? 1'd1;
      unordered_port_0_reg.write_en = std_compareFN_0.done;
      unordered_port_0_reg.in = (std_compareFN_0.unordered == 1'd0) ? 1'd1;
      std_and_0.left = compare_port_0_reg.out;
      std_and_0.right = unordered_port_0_reg.out;
      std_and_1.left = compare_port_0_reg.done;
      std_and_1.right = unordered_port_0_reg.done;
      cmpf_0_reg.in = std_and_0.out;
      cmpf_0_reg.write_en = std_and_1.out;
      std_compareFN_0.go = !std_compareFN_0.done ? 1'd1;
      bb0_0[done] = cmpf_0_reg.done;
    }
```
but somehow `compare_port_0_reg.in = (std_compareFN_0.lt == 1'd0) ? 1'd1` always evaluates to `1` for some reason, even though `std_compareFN_0.left = 32'd1 < std_compareFN_0.right = 32'd9`(actually it's always `1` regardless of input...)

Reply:
okay, `std_reg.in` is marked with `@data`, so guarded assignments like `compare_port_0_reg.in = (std_compareFN_0.lt == 1'd0) ? 1'd1;` will all be rewritten to `compare_port_0_reg.in =  1'd1;`, according to https://docs.calyxir.org/lang/attributes.html?highlight=data#data